### ISSUE TITLE
Serialize `ProgramAst` into files to allow compiling Note Scripts

### ISSUE THREAD

**Issue by Dominik1999** [0xPolygonMiden/miden-vm#1098]
We need to compile note scripts and save them into `miden-lib/assets` directory, see https://github.com/0xPolygonMiden/miden-base/issues/259

One issue here is that we don't have a way to serialize ProgramAst into files. This should be fairly easy to do, and the extension we can use is something like .masb (Miden Assembly binary). This is something we'll need to implement in Miden VM.

**Comment by bobbinth** [comment]
One way to do this is to add `write_to_file()` method on `ProgramAst`. This method would be available only in `std` context. The signature could look like this:
```Rust
pub fn write_to_file<P>(&self, file_path: P) -> io::Result<()>
where
    P: AsRef<Path>,
```

Internally, this could use `ProgramAst::to_bytes()` function for serialization.

**Comment by bobbinth** [comment]
Closed by #1102.
